internal/provider/resource_local_sensitive_file_test.go

```diff
@@ -2,111 +2,57 @@ package provider
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
-	"path"
 	"path/filepath"
-	"runtime"
+	"regexp"
 	"strings"
 	"testing"
 
 	r "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
 func TestLocalSensitiveFile_Basic(t *testing.T) {
 	f := filepath.Join(t.TempDir(), "local_sensitive_file")
 	f = strings.ReplaceAll(f, `\`, `\\`)
 
-	var cases = []struct {
-		path    string
-		content string
-		config  string
-	}{
-		{
-			f,
-			"This is some sensitive content", fmt.Sprintf(`
-				resource "local_sensitive_file" "file" {
-				  content  = "This is some sensitive content"
-				  filename = "%s"
-				}`, f,
-			),
-		},
-		{
-			f,
-			"This is some sensitive base64 content", fmt.Sprintf(`
-				resource "local_sensitive_file" "file" {
-				  content_base64 = "VGhpcyBpcyBzb21lIHNlbnNpdGl2ZSBiYXNlNjQgY29udGVudA=="
-				  filename = "%s"
-				}`, f,
-			),
-		},
-		{
-			f,
-			"This is some sensitive base64 content", fmt.Sprintf(`
-				resource "local_sensitive_file" "file" {
-				  content_base64 = base64encode("This is some sensitive base64 content")
-				  filename = "%s"
-				}`, f,
-			),
+	r.UnitTest(t, r.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+		Steps: []r.TestStep{
+			{
+				Config: testAccConfigLocalSensitiveFileContent("This is some sensitive content", f),
+				Check:  checkFileCreation("local_sensitive_file_resource.test", f),
+			},
+			{
+				Config: testAccConfigLocalSensitiveFileEncodedBase64Content("VGhpcyBpcyBzb21lIGJhc2U2NCBjb250ZW50", f),
+				Check:  checkFileCreation("local_sensitive_file_resource.test", f),
+			},
+			{
+				Config: testAccConfigLocalSensitiveFileDecodedBase64Content("This is some base64 content", f),
+				Check:  checkFileCreation("local_sensitive_file_resource.test", f),
+			},
 		},
-	}
-
-	for i, tt := range cases {
-		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
-			r.UnitTest(t, r.TestCase{
-				Providers: testProviders,
-				Steps: []r.TestStep{
-					{
-						Config: tt.config,
-						Check: func(s *terraform.State) error {
-							content, err := ioutil.ReadFile(tt.path)
-							if err != nil {
-								return fmt.Errorf("config:\n%s\n,got: %s\n", tt.config, err)
-							}
-							if string(content) != tt.content {
-								return fmt.Errorf("config:\n%s\ngot:\n%s\nwant:\n%s\n", tt.config, content, tt.content)
-							}
-							return nil
-						},
-					},
-				},
-				CheckDestroy: checkFileDeleted(tt.path),
-			})
-		})
-	}
+		CheckDestroy: checkFileDeleted(f),
+	})
 }
 
 func TestLocalSensitiveFile_source(t *testing.T) {
 	// create a local file that will be used as the "source" file
-	source_content := "local file content"
-	if err := ioutil.WriteFile("source_file", []byte(source_content), 0644); err != nil {
+	if err := createSourceFile("local file content"); err != nil {
 		t.Fatal(err)
 	}
 	defer os.Remove("source_file")
```

Comment:
Nit: Recommend using https://pkg.go.dev/testing#T.TempDir similar to other tests instead of depending on a static file name/location in the testing directory.